packages/shared/src/hooks/useOnboardingModal.ts

```diff
@@ -83,7 +82,7 @@ export const useOnboardingModal = ({
     registerLocalFilters().then(() => {
       trackEvent({ event_name: AnalyticsEvent.CompleteOnboarding });
       setShouldUpdateFilters(false);
-      modalStateCommand[onboardingVersion]?.(false);
+      setIsOnboardingOpen(false);
```

Comment:
Now that this hook will be instantiated in multiple areas, all of them will be monitoring the `shouldUpdateFilters` and the `registerLocalFilters` will be triggered on all those channels as well. We might need to add a props (like `trackLocalFilters`) in this hook that we need to explicitly pass `true` so when it is `false`, this `useEffect` will return early. Else we will take this out of here and add it in a single-instantiated component (MainLayout or somewhere else).

Reply:
That's a good point, let me check, it should be possible based on the function as it can only be called from one of them.
Another option is to set a query state to state wether it's done?

Reply:
@sshanzel added a state for it, I think that should already catch it.
Let me know what you think.

Reply:
Idea now is to actually split this hook into two
1 will inject the listeners and one the modifiers that way we can ensure the listener is only injected once.

In future mods we should probably move to context, but it's out of scope for now.

Reply:
That's a great idea to isolate what's happening.

Yep, context is always like the best way to go, just feels like overkill sometimes. *redux is waving* lol.